commands/operator-sdk/cmd/scorecard/resource_handler.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorecard
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	proxyConf "github.com/operator-framework/operator-sdk/pkg/ansible/proxy/kubeconfig"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+
+	"github.com/ghodss/yaml"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// createFromYAMLFile will take a path to a YAML file and create the resource. If it finds a
+// deployment, it will add the scorecard proxy as a container in the deployments podspec. If
+// storeKindVersionName is set to true, it will save the kind version and name of the resource
+// in the manifest to the matching global variables (kind, apiversion, and name)
+func createFromYAMLFile(yamlPath string, storeKindVersionName bool) error {
+	yamlFile, err := ioutil.ReadFile(yamlPath)
+	if err != nil {
+		return fmt.Errorf("failed to read file %s: %v", yamlPath, err)
+	}
+	yamlSplit := bytes.Split(yamlFile, []byte("\n---\n"))
+	for _, yamlSpec := range yamlSplit {
+		// some autogenerated files may include an extra `---` at the end of the file
+		if string(yamlSpec) == "" {
+			continue
+		}
+		obj := &unstructured.Unstructured{}
+		jsonSpec, err := yaml.YAMLToJSON(yamlSpec)
+		if err != nil {
+			return fmt.Errorf("could not convert yaml file to json: %v", err)
+		}
+		obj.UnmarshalJSON(jsonSpec)
+		obj.SetNamespace(SCConf.Namespace)
+		if storeKindVersionName {
+			apiversion = obj.GetAPIVersion()
+			kind = obj.GetKind()
+			name = obj.GetName()
+		}
+
+		// dirty hack to merge scorecard proxy into operator deployment; lots of serialization and deserialization
+		if obj.GetKind() == "Deployment" {
+			// TODO: support multiple deployments
+			if deploymentName != "" {
```

Comment:
How does this tell us we have multiple Deployment manifests?
Also I'm wondering what would happen if we add the proxy container and mount the kubeconfig secret onto a non-operator deployment.